proto/mrpb/raft_metadata_repository.go

```diff
@@ -127,7 +127,7 @@ func (crs *LogStreamCommitResults) LastHighWatermark(topicID types.TopicID, hint
 		return crs.CommitResults[i].TopicID >= topicID+1
 	})
 
-	if i > 0 {
+	if i > 0 && crs.GetCommitResults()[i-1].TopicID == topicID {
```

Comment:
For this condition to work, the new topic ID should be higher than the existing one. Have I understood this correctly?

Reply:
I got it. Thanks for your explanation.